### DISCUSSION TITLE
How to point dataset resource URL to an already existing file in file storage?

### DISCUSSION THREAD

**Discussion by kwyz** [ckan/ckan#6170] [Q&A]
In my CKAN environment, I already have an file storage with multiple files in csv format. When I add a new dataset, on resource adding, I need a option to point to an existing file in file storage. By searching in CKAN documentation, I didn't find solution. 
So in few words I'm searching for solution that replace the resource URL with location of file from filestore.
So how to I get it to work as I need?`

**Comment by anuveyatsu** [comment]
@kwyz if you know the URL, you can use it when adding a resource by selecting "Link" instead of "Upload".

**Reply by anuveyatsu** [reply]
@kwyz if you just want to replace the URL of a resource, you can do it either via UI or API. You just need to update `url` attribute of a resource - https://docs.ckan.org/en/2.8/api/index.html#ckan.logic.action.create.resource_create (see `url` parameter). You'd use `resource_patch` API action to update only the `url`.

**Comment by pdekraker-epa** [comment]
I made an extension to help with this sort of situation. We had data stored on a remote system that we wanted accessible by CKAN. It does not add options to the UI, so the resources need to be created via the API. 

https://github.com/USEPA/ckanext-linked-storage
